src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1512,3 +1512,42 @@ def list_consumption_locations():
     client = web_client_factory()
     regions = client.list_geo_regions(sku='Dynamic')
     return [{'name': x.name.lower().replace(" ", "")} for x in regions]
+
+
+def enable_zip_deploy(resource_group_name, name, src, slot=None):
+    # if src is not of .zip extension throw exception
+    if not src.endswith('.zip'):
+        raise CLIError("src '{}' path needs to be .zip file".format(src))
+
+    client = web_client_factory()
+    user_name, password = _get_site_credential(client, resource_group_name, name)
+    scm_url = _get_scm_url(resource_group_name, name, slot)
+    zip_url = scm_url + '/api/zipdeploy'
+
+    import urllib3
+    try:
+        import urllib3.contrib.pyopenssl
+        urllib3.contrib.pyopenssl.inject_into_urllib3()
+    except ImportError:
+        pass
+
+    headers = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(user_name, password))
+    headers['content-type'] = 'application/octet-stream'
+
+    import requests
+    # Read file content
+    fs = open(src, 'rb')
+    zip_content = fs.read()
+
+    try:
+        r = requests.post(zip_url, data=zip_content, headers=headers)
+        if r.status_code != 200:
+            raise CLIError("Zip deployment {} failed with status code '{}' and reason '{}'".format(
```

Comment:
What about `202` is it valid?

Reply:
this will not return 202 - there is an ASYNC equivalent that would

---
In reply to: [145754325](https://github.com/Azure/azure-cli/pull/4700#discussion_r145754325) [](ancestors = 145754325)